File: movies-api/src/main/java/com/ivanfranchin/moviesapi/movie/MovieService.java
package com.ivanfranchin.moviesapi.movie;

import com.ivanfranchin.moviesapi.movie.exception.MovieNotFoundException;
import com.ivanfranchin.moviesapi.movie.model.Movie;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@RequiredArgsConstructor
@Service
public class MovieService {

    private final MovieRepository movieRepository;

    public Movie validateAndGetMovie(String imdbId) {
        return movieRepository.findById(imdbId).orElseThrow(() -> new MovieNotFoundException(imdbId));
    }

    public List<Movie> getMovies() {
        return movieRepository.findAll();
    }

    public Movie saveMovie(Movie movie) {
        return movieRepository.save(movie);
    }

    public void deleteMovie(Movie movie) {
        movieRepository.delete(movie);
    }
}

File: movies-api/src/main/kotlin/com/ivanfranchin/moviesapi/movie/MovieService.kt
package com.ivanfranchin.moviesapi.movie

import com.ivanfranchin.moviesapi.movie.exception.MovieNotFoundException
import com.ivanfranchin.moviesapi.movie.model.Movie
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service

@Service
class MovieService(
    private val movieRepository: MovieRepository
) {

    fun validateAndGetMovie(imdbId: String): Movie {
        return movieRepository.findById(imdbId).orElseThrow {
            MovieNotFoundException(
                imdbId
            )
        }
    }

    fun getMovies(): List<Movie> {
        return movieRepository.findAll()
    }

    fun saveMovie(movie: Movie): Movie {
        return movieRepository.save(movie)
    }

    fun deleteMovie(movie: Movie) {
        movieRepository.delete(movie)
    }
}